File: plot/src/main/java/net/fwitz/math/plot/binary/complex/color/ComplexColorFunction.java
package net.fwitz.math.plot.binary.complex.color;

import net.fwitz.math.complex.Complex;

import java.awt.*;
import java.util.function.BiFunction;

@FunctionalInterface
public interface ComplexColorFunction extends BiFunction<Complex, Complex, Color> {
    /**
     * Calculate the color for the given input and output values.
     *
     * @param c the input value
     * @param z the output value
     * @return the color to paint
     */
    @Override
    Color apply(Complex c, Complex z);
}

File: plot/src/main/kotlin/net/fwitz/math/plot/binary/complex/color/ComplexColorFunction.kt
package net.fwitz.math.plot.binary.complex.color

import net.fwitz.math.binary.complex.Complex
import java.awt.Color

@FunctionalInterface
interface ComplexColorFunction : (Complex, Complex) -> Color {
    /**
     * Calculate the color for the given input and output values.
     *
     * @param c the input value
     * @param z the output value
     * @return the color to paint
     */
    override fun invoke(c: Complex, z: Complex): Color
}